### PR TITLE
feat: popups

### PR DESCRIPTION
**Pull request by dxphilo**

### CODE REVIEW

File: src/components/popups/BasicConfirmAlert.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	text: {
+		type: String,
+		required: true,
+	},
+});
+
+const emit = defineEmits(['close', 'confirm']);
+const showInfoBubble = ref<boolean>(false);
+
+window.addEventListener(`click`, handleCloseClick, false);
```
Comment (ldmweb) [inline src/components/popups/BasicConfirmAlert.vue:14]: We don't need the event listener and its function anymore with the click.stop on the popup overlay

File: src/components/popups/ConfirmPopup.vue [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import { useStoreSettings } from '@/store/settings';
+import InfoIcon from '@/components/icons/Info.vue';
+import CloseIcon from '@/components/icons/XIcon.vue';
+
+const showInfoBubble = ref<boolean>(false);
+const isPosting = ref<boolean>(false);
+const emit = defineEmits(['close', 'post']);
+const settingsStore = useStoreSettings();
+
+window.addEventListener(`click`, handleCloseClick, false);
```
Comment (ldmweb) [inline src/components/popups/ConfirmPopup.vue:12]: same here

File: src/components/popups/Image.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<script setup lang="ts">
+defineProps({
+	image: {
+		type: String,
+		required: true,
+	},
+});
+
+const emit = defineEmits(['close']);
+function closePopup(): void {
```
Comment (ldmweb) [inline src/components/popups/Image.vue:10]: you can directly emit(`close`) in the @click here, and remove the function

File: src/components/popups/Image.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+}
+</script>
+
+<template>
+	<!-- Photo popup -->
+	<div
+		class="popup bg-darkBG dark:bg-gray5 modal-animation fixed top-0 bottom-0 left-0 right-0 z-40 h-screen w-full bg-opacity-50 items-center flex justify-center dark:bg-opacity-50 pt-24"
+		@click="closePopup"
+	>
+		<img :src="image" class="modal-content rounded-lg" />
```
Comment (ldmweb) [inline src/components/popups/Image.vue:21]: Isn't there supposed to be the image Caption here somewhere?

**Review by ldmweb — CHANGES_REQUESTED** [review]

File: src/components/popups/Image.vue (hunk 4 above)
Comment (dxphilo) [reply src/components/popups/Image.vue:21]: > Isn't there supposed to be the image Caption here somewhere?

No we don't from what I see on `capsule-vue`. Can we add it if necessary?

**Review by ldmweb — CHANGES_REQUESTED** [review]
There shouldn't be any shadow on the featuredPhoto popup + the X button should be entered (check how it is on Capsule-vue)
Capture d’écran 2022-09-12 à 17 04 51

On the other 2 popups, the width is missing, usually we put this size on popups:
Capture d’écran 2022-09-12 à 17 09 02

**Comment by dxphilo** [comment]
> There shouldn't be any shadow on the featuredPhoto popup + the X button should be centered (check how it is on Capsule-vue) Capture d’écran 2022-09-12 à 17 04 51
> 
> On the other 2 popups, the width is missing, usually we put this size on popups: Capture d’écran 2022-09-12 à 17 09 02

Resolved. Also if approved I was considering removing the test code from the `post` page since we don't necessarily need them there.

File: src/components/popups/BasicConfirmAlert.vue [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+<template>
+	<div
+		class="popup w-full bg-darkBG dark:bg-gray5 modal-animation fixed top-0 bottom-0 left-0 right-0 z-30 p-5 flex h-full w-full items-center justify-center bg-opacity-25 dark:bg-opacity-25"
+		@click.stop="$emit(`close`)"
+	>
+		<!-- Container -->
+		<section>
+			<div
+				class="w-full bg-lightBG dark:bg-darkBGStop card-animation z-10 overflow-y-auto rounded-lg p-6 pt-4 shadow-lg"
```
Comment (ldmweb) [inline src/components/popups/BasicConfirmAlert.vue:23]: `w-full lg:w-600` is missing here

File: src/components/popups/BasicConfirmAlert.vue [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	},
+});
+
+defineEmits(['close', 'confirm']);
+const showInfoBubble = ref<boolean>(false);
+</script>
+
+<template>
+	<div
+		class="popup w-full bg-darkBG dark:bg-gray5 modal-animation fixed top-0 bottom-0 left-0 right-0 z-30 p-5 flex h-full w-full items-center justify-center bg-opacity-25 dark:bg-opacity-25"
```
Comment (ldmweb) [inline src/components/popups/BasicConfirmAlert.vue:17]: there is two `w-full` here

File: src/components/popups/ConfirmPopup.vue [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+		return;
+	}
+	emit(`post`);
+	isPosting.value = true;
+}
+</script>
+
+<template>
+	<div
+		class="popup w-full bg-darkBG dark:bg-gray5 modal-animation fixed top-0 bottom-0 left-0 right-0 z-30 flex h-screen w-full items-center justify-center bg-opacity-50 dark:bg-opacity-50"
```
Comment (ldmweb) [inline src/components/popups/ConfirmPopup.vue:23]: there is two `w-full` here

File: src/components/popups/ConfirmPopup.vue [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+	>
+		<!-- Container -->
+		<section>
+			<div
+				class="relative bg-lightBG dark:bg-darkBGStop card-animation z-10 mr-5 overflow-y-auto rounded-lg p-6 pt-4 shadow-lg"
```
Comment (ldmweb) [inline src/components/popups/ConfirmPopup.vue:28]: `w-full lg:w-600` is missing here

**Review by ldmweb — CHANGES_REQUESTED** [review]

**Comment by ldmweb** [comment]
> I was considering removing the test code from the post page since we don't necessarily need them there.
Yes once everything is fixed you can remove the test code!

File: src/components/popups/ConfirmPopup.vue (hunk 8 above)
Comment (dxphilo) [reply src/components/popups/ConfirmPopup.vue:28]: What I get when I apply `lg:w-600` . Screen size. `(2560 by 1600)`

Screenshot 2022-09-13 at 15 58 32

Comment (ldmweb) [reply src/components/popups/ConfirmPopup.vue:28]: Not on the overlay, on the white container

Comment (ldmweb) [reply src/components/popups/ConfirmPopup.vue:28]: `lg:w-600` is just for the white container of the popup, not the entire component

Comment (dxphilo) [reply src/components/popups/ConfirmPopup.vue:28]: Ooooh gotcha

File: src/components/popups/FeaturedPhoto.vue [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+</script>
+
+<template>
+	<Teleport to="#popup">
+		<div
+			class="popup bg-darkBG dark:bg-gray5 modal-animation fixed top-0 bottom-0 left-0 right-0 z-30 flex h-screen w-full items-center justify-center bg-opacity-50 dark:bg-opacity-50"
+			@click.stop="$emit(`close`)"
+		>
+			<!-- Container -->
+			<div class="w-full lg:w-600 max-h-90 dark:bg-darkBGStop card-animation z-10 overflow-y-auto rounded-lg p-6 pt-4">
```
Comment (ldmweb) [inline src/components/popups/FeaturedPhoto.vue:20]: Not sure if we should have the `lg:w-600` on this one as it is supposed to zoom on the featuredPhoto to see it better. Same for the height. can you check on Capsule-vue to see how it is there?

File: src/components/popups/FeaturedPhoto.vue [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+</script>
+
+<template>
+	<Teleport to="#popup">
+		<div
+			class="popup bg-darkBG dark:bg-gray5 modal-animation fixed top-0 bottom-0 left-0 right-0 z-30 h-screen w-full flex flex-col justify-center items-center pt-16 bg-opacity-50 dark:bg-opacity-50"
+			@click.stop="$emit(`close`)"
+		>
+			<!-- Container -->
+			<button class="bg-gray1 dark:bg-gray5 focus:outline-none mb-5 rounded-full p-1" @click="$emit(`close`)">
```
Comment (ldmweb) [inline src/components/popups/FeaturedPhoto.vue:20]: When I try on a little screen, the caption is cut, maybe having a max-height on the photo is necessary?
Capture d’écran 2022-09-14 à 16 15 18

**Review by ldmweb — CHANGES_REQUESTED** [review]

**Pull request merged**